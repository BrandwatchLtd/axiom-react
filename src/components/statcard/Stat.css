:root {
  --cmp-stat-vertical-padding: calc(var(--spacing-grid-size) * 0.5);
  --cmp-stat-spacing: calc(var(--spacing-grid-size) * 2);
}

.ax-stat-cards__stat {
  padding-top: var(--cmp-stat-vertical-padding);
  padding-bottom: var(--cmp-stat-vertical-padding);
  list-style: none;

  & + .ax-stat-cards__stat {
    margin-left: var(--cmp-stat-spacing);
    padding-left: var(--cmp-stat-spacing);
    border-left: var(--component-border-width) solid var(--color-theme-border);
  }
}

.ax-stat-cards__stat--tiny-clanger { color: var(--color-product-tiny-clanger); }
.ax-stat-cards__stat--critical-mass { color: var(--color-product-critical-mass); }
.ax-stat-cards__stat--paradise-lost { color: var(--color-product-paradise-lost); }
.ax-stat-cards__stat--serene-sea { color: var(--color-product-serene-sea); }
.ax-stat-cards__stat--giant-leap { color: var(--color-product-giant-leap); }
.ax-stat-cards__stat--moon-lagoon { color: var(--color-product-moon-lagoon); }
.ax-stat-cards__stat--terra-form { color: var(--color-product-terra-form); }
.ax-stat-cards__stat--primeval-soup { color: var(--color-product-primeval-soup); }
.ax-stat-cards__stat--new-horizon { color: var(--color-product-new-horizon); }
.ax-stat-cards__stat--blast-off { color: var(--color-product-blast-off); }
.ax-stat-cards__stat--ground-control { color: var(--color-product-ground-control); }
.ax-stat-cards__stat--luna-dust { color: var(--color-product-luna-dust); }
