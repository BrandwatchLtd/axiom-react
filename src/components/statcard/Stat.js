import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../base/Base';
import Heading from '../typography/Heading';
import './Stat.css';

export default class Stat extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    color: PropTypes.oneOf([
      'tiny-clanger',
      'critical-mass',
      'paradise-lost',
      'serene-sea',
      'giant-leap',
      'moon-lagoon',
      'terra-form',
      'primeval-soup',
      'new-horizon',
      'blast-off',
      'ground-control',
      'luna-dust',
    ]),
  };

  render() {
    const { children, color, ...rest } = this.props;
    const classes = classnames('ax-stat-cards__stat', {
      [`ax-stat-cards__stat--${color}`]: color,
    });

    return (
      <Base { ...rest }
          Component="li"
          className={ classes }>
        <Heading textSize="headline">{ children }</Heading>
      </Base>
    );
  }
}
