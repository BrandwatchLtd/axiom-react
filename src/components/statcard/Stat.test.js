import React from 'react';
import renderer from 'react-test-renderer';
import Stat from './Stat';

const getComponent = (props = {}) =>
  renderer.create(
    <Stat { ...props }>Lorem</Stat>
  );

describe('Stat', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with color', () => {
    [
      'rose',
      'pink',
      'purple',
      'lilac',
      'blue',
      'teal',
      'green',
      'chartreuse',
      'amber',
      'orange',
      'brown',
      'grey',
    ].forEach((color) => {
      it(color, () => {
        const component = getComponent({ color });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
