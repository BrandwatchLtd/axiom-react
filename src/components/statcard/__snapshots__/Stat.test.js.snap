// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat renders with color amber 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--amber"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color blue 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--blue"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color brown 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--brown"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color chartreuse 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--chartreuse"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color green 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--green"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color grey 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--grey"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color lilac 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--lilac"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color orange 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--orange"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color pink 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--pink"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color purple 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--purple"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color rose 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--rose"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with color teal 1`] = `
<li
  className="ax-stat-cards__stat ax-stat-cards__stat--teal"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;

exports[`Stat renders with defaultProps 1`] = `
<li
  className="ax-stat-cards__stat"
>
  <h3
    className="ax-heading ax-heading--headline ax-space--tiny"
  >
    <span
      className="ax-text--weak"
    >
      Lorem
    </span>
  </h3>
</li>
`;
